.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,65 @@
+name: Build and Publish Marketplace
+on:
+  push:
+
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    container: cimg/go:1.19
+    defaults:
+      run:
+        shell: bash
+    steps:
+    - name: build/checkout-repo
+      uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+      with:
+        fetch-depth: 0
+    - name: build/ensure-master-exists
+      run: git branch master origin/master
+    - name: build/check-style
+      run: make check-style
+    - name: build/test
+      run: make test
+    - name: build/build
+      run: make build
+    - name: build/persist-build
+      uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+      with:
+        name: marketplace-build
+        path: dist/
+        retention-days: 2
+
+  deploy:
+    if: ${{ github.repository == 'mattermost/mattermost' && (github.ref_name == 'master' || github.ref_name == 'production') }}
```

Comment:
We used to use other branches as well IIRC, though this is probably fine

Reply:
For managing the deploys, you mean?
My understanding of the [CircleCI config's deploy definition](https://github.com/mattermost/mattermost-marketplace/pull/393/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L60-L71) only those two branches will trigger the respective staging and production deploys.

Reply:
@mvitale1989 Is right. Deployment should only run from these two branches.